R/tree-subset.R

```diff
@@ -0,0 +1,84 @@
+#' Subset tree objects by related nodes
+#'
+#' This function allows for a tree object to be subset by specifying a
+#' node and returns all related nodes within a selected number of
+#' levels
+#'
+#' @param tree a tree object of class phylo
+#' @param node either a tip label or a node number for the given
+#' tree that will be the focus of the subsetted tree
+#' @param levels_back a number specifying how many nodes back from
+#' the selected node the subsetted tree should include
+#'
+#' @details This function will take a tree and a specified node from
+#' that tree and subset the tree showing all relatives back to a specified
+#' number of nodes. This function allows for a combination of
+#' \link{\code{ancestor}} and \link{\code{offspring}} to return a subsetted
+#' tree that is of class phylo. This allows for easy graphing of the tree
+#' with \code{ggtree}
+#'
+#' @examples
+#' \dontrun{
+#'   nwk <- system.file("extdata", "sample.nwk", package="treeio")
+#'   tree <- read.tree(nwk)
+#'
+#'   sub_tree <- tree_subset(tree, node = "A", levels_back = 3)
+#'   ggtree(sub_tree) + geom_tiplab() + geom_nodelab()
+#' }
+#'
+#' @export
+tree_subset <- function(tree, node, levels_back = 5){
+  # error catching to ensure the tree input is of class phylo
+  if (class(tree) == "phylo") {
+    tree_df <- tidytree::as_data_frame(tree)
+  } else {
+    stop("tree must be of class 'phylo'")
+  }
+
+  # error catching to ensure the levels_back input is numeric
+  # or can be converted to numeric
+  if (!is.numeric(levels_back)) {
+    levels_back <- as.numeric(levels_back)
+    if (is.na(levels_back)) stop("'levels_back' must be of class numeric")
+  }
+
+  # error catching to ensure that treeio is installed
+  if (!requireNamespace("treeio", quietly = TRUE)) {
+    stop("`treeio` is required to use tree_subset. It is not currently installed. ",
+         "You can install it by running either `install.packages('treeio')` for ",
+         "the current CRAN release or `devtools::install_github('GuangchuangYu/treeio')` ",
+         "for the development version.",
+         call. = FALSE)
+  }
+
```

Comment:
this function should goes to `treeio` pkg, as it depends on treeio and the output is also a tree object instead of tbl_tree.

Reply:
another reason is that treeio depends on tidytree and tidytree can't depends on treeio, otherwise there will be a circle dependency and all of these pkgs will fail to compile and install.